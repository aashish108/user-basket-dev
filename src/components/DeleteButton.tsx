import styled from '@emotion/styled';
import { useContext, useState } from 'react';
import { Context } from '../contexts/Context';

const Button = styled.button`
 background-color: red;
 color: white;
 padding: 10px;
 &:hover {
  background-color: darkred;
 }
`;

type A = { name: string };

const DeleteButton = ({ name }: A) => {
 const { basketItems, setBasketItems } = useContext(Context);
 const [currentProductName] = useState(name);
 const deleteProduct = () => {
  const newBasket = basketItems.filter(
   (item) => item.productName !== currentProductName
  );
  setBasketItems(newBasket);
 };
 return <Button onClick={deleteProduct}>Delete</Button>;
};

export default DeleteButton;
