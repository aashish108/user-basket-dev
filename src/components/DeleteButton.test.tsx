import { render, screen } from '@testing-library/react';
import DeleteButton from './DeleteButton';

it('renders', async () => {
 render(<DeleteButton name="Apple" />);
 const text = await screen.findAllByText(/Delete/);
 expect(text).toBeTruthy();
});
