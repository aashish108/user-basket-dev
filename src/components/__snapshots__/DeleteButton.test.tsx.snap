// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="css-1c5dypr"
  onClick={[Function]}
>
  Delete
</button>
`;

exports[`shows the correct button text 1`] = `
<button
  className="css-1c5dypr"
  onClick={[Function]}
>
  Delete
</button>
`;
